pkg/userdata/coreos/userdata.go

```diff
@@ -197,6 +197,7 @@ systemd:
           --cluster-dns={{ ipSliceToCommaSeparatedString .ClusterDNSIPs }} \
           --cluster-domain=cluster.local \
           --authentication-token-webhook=true \
+          --hostname-override={{ .MachineSpec.Name }} \
```

Comment:
This is to avoid failures like this one: `Error updating node status, will retry: error getting node "scenario-2-ubuntu-docker-17": nodes "scenario-2-ubuntu-docker-17" not found`

If a cloud provider is configured, this wont have any effect [according to this comment](https://github.com/kubernetes/kubernetes/issues/22984#issuecomment-196909977)